### DISCUSSION TITLE
Dependency graph in the dashboard

### DISCUSSION THREAD

**Discussion by not7cd** [renovatebot/renovate#27245] [Suggest an Idea]
### Tell us more.

As of now, Renovate creates an issue that has a list of all detected dependencies and their files.

While developing pip-compile I found that writing dependency graphs is quite easy with Mermaid.js. For now I create them as debug logs. With native support in GH, such graphs could be included in managers folded section of the dependency dashboard.

Example

```
DEBUG: pip-compile: dependency graph:
       graph TD
         requirements-dev.txt[[requirements-dev.txt]]
         requirements.txt[[requirements.txt]]
         requirements-dev.in --> requirements-dev.txt
         requirements.in --> requirements.txt
         requirements.txt --> requirements-dev.in
        (repository=icetek-lab/renovate-case2)

```

```mermaid
       graph TD
         requirements-dev.txt[[requirements-dev.txt]]
         requirements.txt[[requirements.txt]]
         requirements-dev.in --> requirements-dev.txt
         requirements.in --> requirements.txt
         requirements.txt --> requirements-dev.in
      
```

**Comment by viceice** [comment]
the problem is the size, there are a lot projects which are at issue body limits

**Reply by HonkingGoose** [reply]
Here are some issues that also mention the "issue body character limits":

- #17174
- #17749
- #15850
